osp/core/session/sparql_backend.py

```diff
@@ -91,8 +98,36 @@ def __getitem__(self, variable_name):
 
         Handle wrapper IRIs.
         """
-        x = self._get(variable_name)
-        if x is not None and x.startswith(CUDS_IRI_PREFIX) \
-                and uid_from_iri(x) == uuid.UUID(int=0):
-            return iri_from_uid(self.session.root)
-        return x
+        iri = self._get(variable_name)
+        if iri is not None and iri.startswith(CUDS_IRI_PREFIX) \
+                and uid_from_iri(iri) == uuid.UUID(int=0):
+            iri = iri_from_uid(self.session.root)
+        return self._check_datatype(variable_name, iri)
+
+    def _check_datatype(self, variable_name, iri):
+        """Check if iri shall be converted to a certain datatype."""
+        if iri:
+            try:
+                variable_type = self.datatypes[variable_name]
+                if variable_type == 'cuds':
+                    cuds_query = self.session.load_from_iri(iri)
+                    return cuds_query.first()
+                elif variable_type in [float, str, int, bool]:
+                    return variable_type(iri._value)
+                elif callable(variable_type):
+                    return variable_type(iri)
+                else:
+                    raise TypeError("Variable type not understood")
+            except KeyError:
+                return iri
+            except Exception as excep:
+                raise ValueError(excep)
+        else:
+            return iri
+
+
+class SparqlDataTypes(dict):
```

Comment:
If no additional use comes to your mind, I would just use plain simple dictionary.